pkg/util/constants.go

```diff
@@ -88,8 +89,25 @@ var rng struct {
 // strconv.ParseBool then metamorphic testing will not be enabled.
 const DisableMetamorphicEnvVar = "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING"
 
+// Returns true iff the current process is eligible to enable metamorphic
+// variables. When run under Bazel, checking if we are in the Go test wrapper
+// ensures that metamorphic variables are not initialized and logged twice
+// from both the wrapper and the main test process, as both will perform
+// initialization of the test module and its dependencies.
+func metamorphicEligible() bool {
+	if !buildutil.CrdbTestBuild {
+		return false
+	}
+
+	if bazel.TestByBazel() && bazel.TestWrapped() {
```

Comment:
Why do we need both predicates? Couldn't this be simplified to just use `TestWrapped`? This way there is no need for `metamorphicEligible`, 

```
func init() {
   if buildutil.CrdbTestBuild && !bazel.TestWrapped {
   ...    
}
```

Reply:
I agree that it seems we don't need two different functions since one doesn't make sense without the other. It seems we can have just one:

```
// bazel.go
func checkEnvBool(name string) {
	if env, ok := os.LookupEnv(name); ok {
		if val, err := strconv.ParseBool(env); err == nil {
			return val
		}
	}
  return false
}

func IsBazelTest() {
    return checkEnvBool("BAZEL_TEST") && !checkEnvBool("GO_TEST_WRAP")
}

// constants.go
func metamorphicEligible() {
   return (bazel.BuiltWithBazel() && bazel.IsBazelTest()) || buildutil.CrdbTestBuild
}
```

Reply:
Stan's suggested method works too, as long as `TestWrapped()` (function should still be renamed) returns `false` for tests running outside of Bazel, which it should.